packages/builders/src/interactions/commands/contextMenu/ContextMenuCommand.ts

```diff
@@ -0,0 +1,142 @@
+import type {
+	ApplicationCommandType,
+	ApplicationIntegrationType,
+	InteractionContextType,
+	LocaleString,
+	Permissions,
+	RESTPostAPIContextMenuApplicationCommandsJSONBody,
+} from 'discord-api-types/v10';
+import type { RestOrArray } from '../../../util/normalizeArray.js';
+import { normalizeArray } from '../../../util/normalizeArray.js';
+import { CommandBuilder } from '../Command.js';
+
+/**
+ * The type a context menu command can be.
+ */
+export type ContextMenuCommandType = ApplicationCommandType.Message | ApplicationCommandType.User;
+
+/**
+ * A builder that creates API-compatible JSON data for context menu commands.
+ */
+export abstract class ContextMenuCommandBuilder extends CommandBuilder<RESTPostAPIContextMenuApplicationCommandsJSONBody> {
```

Comment:
I'd say something like `SharedName` -> `SharedNameAndDescription` (because there's places where we want only name, others where we want both), `Command` gets some of the common stuff between context menu/chat inputs (nsfw etc).